### PR TITLE
Scripts/Spells: Fix mage spell Ray of Frost

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Fix mage talent "Ray of Frost"

**Tests performed:**

Tested in-game

### CODE REVIEW

File: src/server/scripts/Spells/spell_mage.cpp [hunk 1]
```diff
@@ -1110,6 +1112,52 @@ class spell_mage_prismatic_barrier : public AuraScript
...
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_MAGE_RAY_OF_FROST_BONUS });
+    }
+
+    void HandleEffectPeriodic(AuraEffect const* aurEff)
+    {
+        if (Unit* caster = GetCaster())
+        {
+            if (aurEff->GetTickNumber() > 1) // First tick should deals base damage
```
Comment (Nyr97) [inline src/server/scripts/Spells/spell_mage.cpp:1150]: typo

**Review by Nyr97 — CHANGES_REQUESTED** [review]

File: src/server/scripts/Spells/spell_mage.cpp [hunk 2]
```diff
@@ -1110,6 +1112,52 @@ class spell_mage_prismatic_barrier : public AuraScript
...
+class spell_mage_ray_of_frost : public SpellScript
+{
+    PrepareSpellScript(spell_mage_ray_of_frost);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_MAGE_RAY_OF_FROST_FINGERS_OF_FROST });
+    }
+
+    void HandleFingersOfFrost()
```
Comment (mdX7) [inline src/server/scripts/Spells/spell_mage.cpp:1125]: would recommend naming it similar to the hook, like `HandleOnHit`

File: src/server/scripts/Spells/spell_mage.cpp (hunk 1 above)
Comment (aquadeus) [reply src/server/scripts/Spells/spell_mage.cpp:1150]: where?

Comment (Faq) [reply src/server/scripts/Spells/spell_mage.cpp:1150]: or just "First tick deals base damage"

**Comment by Shauren** [comment]
You need to remove SPELL_MAGE_RAY_OF_FROST_FINGERS_OF_FROST when channel gets interrupted
Don't use hotfix db to edit spells, put that in SpellMgr

**Comment by aquadeus** [comment]
> You need to remove SPELL_MAGE_RAY_OF_FROST_FINGERS_OF_FROST when channel gets interrupted Don't use hotfix db to edit spells, put that in SpellMgr

Ok changed

File: src/server/scripts/Spells/spell_mage.cpp [hunk 3]
```diff
@@ -1110,6 +1112,59 @@ class spell_mage_prismatic_barrier : public AuraScript
...
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_MAGE_RAY_OF_FROST_BONUS, SPELL_MAGE_RAY_OF_FROST_FINGERS_OF_FROST });
+    }
+
+    void HandleEffectPeriodic(AuraEffect const* aurEff)
+    {
+        if (Unit* caster = GetCaster())
+        {
+            if (aurEff->GetTickNumber() > 1) // First tick should deal base damage
```
Comment (Seyden) [inline src/server/scripts/Spells/spell_mage.cpp:1150]: aren't you like 1 tick too late? i mean when the 2nd tick happenend, only then you apply the bonus

Comment (mdX7) [reply src/server/scripts/Spells/spell_mage.cpp:1150]: the tick number is increased before the hook is called, so its 1 for the first tick - not 0.
Ref:
https://github.com/TrinityCore/TrinityCore/blob/2a5329847636ddc080c461bda06375edfbf3815d/src/server/game/Spells/Auras/SpellAuraEffects.cpp#L1027-L1036

Comment (Seyden) [reply src/server/scripts/Spells/spell_mage.cpp:1150]: That doesn’t answer my question.. ticksDone starts at 0 and is increased to 1 and then the hook is called and GetTickNumber returns 1 and not 2.

Therefore you still need a second call in order to get this bonus or am I seeing things ?

Comment (mdX7) [reply src/server/scripts/Spells/spell_mage.cpp:1150]: This hook is executes before dmg is done. So the check is fine.

Ref
https://github.com/TrinityCore/TrinityCore/blob/2a5329847636ddc080c461bda06375edfbf3815d/src/server/game/Spells/Auras/SpellAuraEffects.cpp#L1063

**Pull request merged**